code_challenges/code_wars/tic_tac_toe_checker.py

```diff
@@ -0,0 +1,31 @@
+def find_winner(board):
+
+    ways_to_win = {
+        0: [board[0][0], board[0][1], board[0][2]],
+        1: [board[0][0], board[1][0], board[2][0]],
+        2: [board[0][1], board[1][1], board[2][1]],
+        3: [board[0][2], board[1][2], board[2][2]],
+        4: [board[0][0], board[1][1], board[2][2]],
+        5: [board[1][0], board[1][1], board[1][2]],
+        6: [board[2][0], board[2][1], board[2][2]],
+        7: [board[0][2], board[1][1], board[2][0]],
+
+    }
+    for moves in ways_to_win:
+        if ways_to_win[moves] == [1, 1, 1]:
+            return 1
+        elif ways_to_win[moves] == [2, 2, 2]:
+            return 2
+    for moves in ways_to_win:
+        if 0 in ways_to_win[moves]:
+            return -1
+    return 0
```

Comment:
i chose a dictionary because it is faster. it is also more visually appealing when i was counting how many ways to win there were.

Reply:
> i chose a dictionary because it is faster

_Dictionary are faster when you are not iterating over keys._ `for key in dictionary` is the same speed as `for item in list`, not faster.

> it is also more visually appealing

This is a good factor, if it helps think about the code or organize the code it might be a better choice than a list.


In any case `for moves in ways_to_win:` isn't really the best variable naming because `moves` is a number.  I'd suggest `for way_number in ways_to_win:`, but even better would be `for value in dictionary.values()` like:
```python
    for way in ways_to_win.values():
        if way == [1, 1, 1]:
            return 1
        elif way == [2, 2, 2]:
            return 2
    for way in ways_to_win.values():
        if 0 in way:
            return -1
    return 0
```